x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/policies/package_policies.tsx

```diff
@@ -95,6 +95,19 @@ export const PackagePoliciesPage = ({ name, version }: PackagePoliciesPanelProps
 
   const agentEnrollmentFlyoutExtension = useUIExtension(name, 'agent-enrollment-flyout');
 
+  // Handle the "add agent" link displayed in post-installation toast notifications in the case
+  // where a user is clicking the link while on the package policies listing page
+  useEffect(() => {
+    history.listen((location) => {
```

Comment:
it looks like we should probably also unlisten?

Reply:
Yep good call. I updated this `useEffect` to call the `unlisten` callback as its cleanup operation in https://github.com/elastic/kibana/pull/104493/commits/6f405bd4a87a75680c224ace034f4831507fdfbe.